### ISSUE TITLE
Typescript S4328 does not support aliases

### ISSUE THREAD

**Issue by tremblaysimon** [SonarSource/SonarJS#1851]
**I want to report a bug.**

**SonarJS version:** 6.1.0.11503<br>

**Rule key:** typescript:S4328

**Reproducer** 

If I use aliases with import of internal modules the S4328 rule is raised for those imports.

**Example**

tsconfig.json
```
{
  "compileOnSave": false,
  "compilerOptions": {
    "baseUrl": "./",
    "paths": {
      "$core/*": ["src/app/core/*"],
...
```
in sometypescript.ts
```
import { ConfigurationService } from '$core/services/configuration.service';
```

**Expected behavior**

It should not raise S4328 rule for that import since it is not an external module.

---

Reading that ROADMAP here: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/ROADMAP.md, it seems that no-implicit-dependencies from tslint has been implemented in the eslint-plugin-import: https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-extraneous-dependencies.md

It seems that aliases are supported in that rule. Does it make sense to use that plugin instead for that kind of rule?

**Comment by cschuff** [comment]
Facing similar problems. I have discovered this:

> Dependencies should be explicit
> 
> Dependencies should be explicitly listed in the package.json file. Importing a module that is not declared as a dependency makes it an implicit one and is bound to create problems.
> 
> Parameters
> whitelist	
> Comma separated list of modules to ignore while checking in package.json

There seems to be a `whitelist` parameter that allows for making exceptions to that rule. But I have not yet figured out how to set it...

Anyone?

**Comment by Shinigami92** [comment]
What needs to be done that `sonar-scanner-npm` can benefit from this fix?

**Comment by claireqian1007** [comment]
<img width="833" alt="image" src="https://user-images.githubusercontent.com/41279223/203695695-7da35987-c921-4910-a26c-fea7b92d0a9d.png">

can try to add specific path in ```sonar.sources``` like this:
```
// in my tsconfig
"paths": {
      "src": [
        "./src/*"
      ],
    },

// in my sonar-project.properties
sonar.sources=src/components,src/Navigation,src/screens,src/store,src/types,src/utils
```
it works for me, in sonar scan report 'dependency should be explicit' issues are removed

**Comment by last-Programmer** [comment]
We are having this issue in our project. Sonarqube is not resepcting paths in tsconfig and it is always reporting S4328 error,

**Comment by theanlay** [comment]
I got Issue on using aliases to import of internal modules. after I add path "@/*": ["./src/*"],
while import internal modules got error 
Either remove this import or add it as a dependency. sonarlint(typescript:S4328).
I'm using Vue3, Vite with TypeScript.

**Comment by thendricks0** [comment]
Still happens with NuxtJS projects https://community.sonarsource.com/t/s4328-false-positive-on-nuxtjs-projects-with-path-aliases/135027
